extensions/amp-action-macro/0.1/amp-action-macro.js

```diff
@@ -87,6 +88,18 @@ export class AmpActionMacro extends AMP.BaseElement {
     // We want the macro to be available wherever it is in the document.
     return true;
   }
+
+  /**
+   * Checks if the action is triggering itself.
+   * @param {!../../../src/service/action-impl.ActionInvocation} invocation
+   * @return {boolean}
+   * @private
+   */
+  isRecursiveInvocation_(invocation) {
+    return invocation.source.getAttribute('id')
+        === this.element.getAttribute('id')
+        && invocation.source.tagName === 'AMP-ACTION-MACRO';
```

Comment:
We need to worry about cycles generally. I.e. not just `x -> x` but `x -> y -> x`.